chaosaws/__init__.py

```diff
@@ -56,43 +57,70 @@ def aws_client(resource_name: str, configuration: Configuration = None,
     as we do not read those settings from the `secrets` object.
     """  # noqa: E501
     configuration = configuration or {}
-    region = configuration.get("aws_region", "us-east-1")
-    creds = get_credentials(secrets)
+    aws_profile_name = configuration.get("aws_profile_name")
     aws_assume_role_arn = configuration.get("aws_assume_role_arn")
+    params = get_credentials(secrets)
+
+    region = configuration.get("aws_region")
+    if not region:
+        logger.debug(
+            "The configuration key `aws_region` is not set, looking in the "
+            "environment instead for `AWS_REGION` or `AWS_DEFAULT_REGION`")
```

Comment:
This block seems a little kludgy. Could be simplified like:

`region = configuration.get('aws_region') or os.getenv('AWS_REGION', os.getenv('AWS_DEFAULT_REGION'))`

And then you only need to have a single `if` statement to determine the region...

Also, if there is no region specified here, why does it not stop processing?

Reply:
True on both accounts.

On the former point, I wanted to help users figure our they have various options to configure their region but that might be more confusing?

On the latter point, it could be the right option indeed to avoid bizarre side effects by running when we know we cannot.

Reply:
I am not interrupting the execution indeed but I've left the intermediate message which hopefully can educate when a region was not set. I like when log message teaches you how to fix a problem :)